### PR TITLE
Add custom subdomain

### LINKED ISSUE #5327: [DOCS] deploy docs in custom subdomain

### PR DESCRIPTION
**Pull request by davidberenstein1957**
# Description

Closes #5327 

**Type of change**

- Documentation update

**How Has This Been Tested**

**Checklist**

- I added relevant documentation
- I followed the style guidelines of this project
- I did a self-review of my code
- I made corresponding changes to the documentation
- I confirm My changes generate no new warnings
- I have added tests that prove my fix is effective or that my feature works
- I have added relevant notes to the CHANGELOG.md file (See https://keepachangelog.com/)

### CODE REVIEW

**Comment by davidberenstein1957** [comment]
After merging, we should update the DNS

File: argilla-frontend/nuxt.config.ts [hunk 1]
```diff
@@ -203,15 +203,15 @@ const config: NuxtConfig = {
 
   publicRuntimeConfig: {
     clientVersion: pkg.version,
-    documentationSite: "https://docs.argilla.io/",
+    documentationSite: "https://docs.v1.argilla.io/",
```
Comment (frascuchon) [inline argilla-frontend/nuxt.config.ts]: Should be the `https://docs.v1.argilla.io/` reachable?

**Comment by frascuchon** [comment]
Maybe [this one](https://github.com/argilla-io/argilla/pull/5321) is related

File: argilla-frontend/nuxt.config.ts (hunk 1 above)
Comment (davidberenstein1957) [reply argilla-frontend/nuxt.config.ts]: Yes it should be after the DNS updates.

**Comment by frascuchon** [comment]
I think this PR should be merged after the publishing of the release (at least changes related to the docs) What do you think? Could we let UI links pointing the current `argilla-io.....` domain and let github make the redirection? (Just to apply these changes after the release)

**Comment by davidberenstein1957** [comment]
> I think this PR should be merged after the publishing of the release (at least changes related to the docs) What do you think? Could we let UI links pointing the current `argilla-io.....` domain and let github make the redirection? (Just to apply these changes after the release)

I don't agree, @frascuchon because some of these changes also need to be represented in the product and Python code.

**Comment by davidberenstein1957** [comment]
@frascuchon is this fixing the tests? https://github.com/argilla-io/argilla/pull/5330

File: argilla-frontend/nuxt.config.ts [hunk 2]
```diff
@@ -203,15 +203,15 @@ const config: NuxtConfig = {
...
+      "https://docs.v1.argilla.io/en/latest/reference/webapp/features.html#semantic-search",
     documentationSiteLabelScheme:
-      "https://docs.argilla.io/en/latest/guides/log_load_and_prepare_data.html#define-a-labeling-schema",
+      "https://docs.v1.argilla.io/en/latest/guides/log_load_and_prepare_data.html#define-a-labeling-schema",
     documentationSiteQueryDatasets:
-      "https://docs.argilla.io/en/latest/guides/query_datasets.html",
+      "https://docs.v1.argilla.io/en/latest/guides/query_datasets.html",
     documentationPersistentStorage:
-      "https://docs.argilla.io/en/latest/getting_started/installation/deployments/huggingface-spaces.html#setting-up-persistent-storage",
+      "https://docs.v1.argilla.io/en/latest/getting_started/installation/deployments/huggingface-spaces.html#setting-up-persistent-storage",
```
Comment (frascuchon) [inline argilla-frontend/nuxt.config.ts]: If the product points to the new docs, Should not be the docs.argilla.io?

File: argilla-server/src/argilla_server/_app.py [hunk 3]
```diff
@@ -171,7 +171,7 @@ def show_telemetry_warning():
         message += inspect.cleandoc(
             "Argilla uses telemetry to report anonymous usage and error information. You\n"
             "can know more about what information is reported at:\n\n"
-            "    https://docs.argilla.io/en/latest/reference/telemetry.html\n\n"
+            "    https://docs.v1.argilla.io/en/latest/reference/telemetry.html\n\n"
```
Comment (frascuchon) [inline argilla-server/src/argilla_server/_app.py]: why does this point to the old documentation?

File: argilla/docs/CNAME [hunk 4]
```diff
@@ -0,0 +1 @@
+docs.v1.argilla.io
```
Comment (frascuchon) [inline argilla/docs/CNAME]: should not be the `docs.argilla.io` ?

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -17,5 +17,5 @@ We understand that getting started might be a bit difficult, therefore, you can
 
 ## Want to work on your own?
 
-For more seasoned contributors, we recommend taking a look at the [contributor section](https://docs.argilla.io/en/latest/community/contributing.html) in our docs.
+For more seasoned contributors, we recommend taking a look at the [contributor section](https://docs.v1.argilla.io/en/latest/community/contributing.html) in our docs.
```
Comment (frascuchon) [inline CONTRIBUTING.md]: docs.argilla.io?

**Review by dvsrepo — CHANGES_REQUESTED** [review]
I don't have additional comments to those of @frascuchon

**Comment by frascuchon** [comment]
> @frascuchon is this fixing the tests? #5330

Tests were fixed in https://github.com/argilla-io/argilla/pull/5331

**Review by frascuchon — APPROVED** [review]

**Pull request merged**